dags/veda_data_pipeline/groups/discover_group.py

```diff
@@ -42,60 +43,59 @@ def get_files_to_process(ti):
     """Get files from S3 produced by the discovery task.
     Used as part of both the parallel_run_process_rasters and parallel_run_process_vectors tasks.
     """
-    payload = ti.xcom_pull(task_ids=f"{group_kwgs['group_id']}.discover_from_s3")
-    payloads_xcom = payload.pop("payload", [])
+    dynamic_group_id = ti.task_id.split(".")[0]
+    payload = ti.xcom_pull(task_ids=f"{dynamic_group_id}.discover_from_s3")
+    payloads_xcom = payload[0].get("payload", [])
```

Comment:
The veda_dataset_pipeline DAG is working as expected (caveats in a different issue) but now I am no longer to trigger veda_discover. I.e. 
```
    "collection": "OMI_trno2-COG-deleteme",
		"bucket": "veda-data-store-staging",
    "datetime_range": "year",
    "discovery": "s3",
    "filename_regex": "^(.*).tif$",
    "prefix": "OMI_trno2-COG/"
}
```
results in 
```
AIRFLOW_CTX_DAG_RUN_ID=manual__2024-07-09T19:58:28+00:00
[2024-07-09, 19:58:59 UTC] {{taskinstance.py:1768}} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 75, in wrapper
    return func(*args, session=session, **kwargs)
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow_multi_dagrun/operators.py", line 28, in execute
    for conf in self.python_callable(*self.op_args, **self.op_kwargs):
  File "/usr/local/airflow/dags/veda_data_pipeline/groups/discover_group.py", line 48, in get_files_to_process
    payloads_xcom = payload[0].get("payload", [])
KeyError: 0
```

Reply:
I haven't looked at the change but a quick test in SIT the error seems to have moved a few lines down in discover_group
```
[2024-07-12, 14:27:24 UTC] {{taskinstance.py:1768}} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 75, in wrapper
    return func(*args, session=session, **kwargs)
  File "/usr/local/airflow/.local/lib/python3.10/site-packages/airflow_multi_dagrun/operators.py", line 28, in execute
    for conf in self.python_callable(*self.op_args, **self.op_kwargs):
  File "/usr/local/airflow/dags/veda_data_pipeline/groups/discover_group.py", line 57, in get_files_to_process
    **payload[0],
```